numpy/core/tests/test_multiarray.py

```diff
@@ -2713,6 +2713,20 @@ def test_arr_mult(self, func):
             assert_equal(func(edf, edf.T), eddtf)
             assert_equal(func(edf.T, edf), edtdf)
 
+    @pytest.mark.parametrize('func', (np.dot, np.matmul))
+    def test_no_dgemv(self, func):
```

Comment:
Any reason to just check doubles?

Reply:
I'm thinking `f, d, c, z`, maybe separately for the two inputs. That way can also test the promotion. The advantage is that if the code gets refactored and introduces new code paths. Maybe those cases are already tested somewhere for the usual case?

Reply:
It is difficult to test blas functionality
- we have no external indication whether blas is used or not
- we do not get a warning/error if blas is called incorrectly (the `xerblas` mechanism is not working correctly). Perhaps it is worthwhile opening an issue for this but the working assumption is we cannot call blas incorrectly.

I determined that my fix works by manually setting a breakpoint at the blas calls and verifying, via `runtests.py -vv` and manually examining dimensions and strides that the calls were correct, and by noticing that xerblas error messages do not appear on stderr.

While it would be simple to add the other blas-able cases, what would the expected error vs. non-error output be? It is not clear to me there is added value in quadrupling the test.

Reply:
Since I was expanding the tests anyway I added the other BLAS dtypes and `int` as well for good measure.

Reply:
> I added the other BLAS dtypes and `int` as well for good measure

I see you requested **combinations** of dtypes. This is covered in other tests of the ufunc machinery, that it promotes and copies input where needed.